### DISCUSSION TITLE
Unable to pull DXCC data from CLubLog after installation

### DISCUSSION THREAD

**Discussion by 4O4A** [magicbug/Cloudlog#1961] [General]
Hi guys,

I have a few problems with brand new installation of ClubLog.
First, when download it from https://github.com/magicbug/Cloudlog/archive/refs/heads/master.zip version still shows 1.7.
I am not sure am I downloading correct version or not?

When I download files from https://github.com/magicbug/Cloudlog/releases (say, version 2.3) config.sample.php file reads version 2.1 but after the installation version in /application/config/config.php reads 1.7.

More or less, not a big deal.

More serious issue is with my updates of DXCC from ClubLog.
Last 3-4 installations I never managed to download data from ClubLog.
I am not sure is it problem of ClubLog, Cloudlog, my installations, folder permissions or what.
I solved it last time by copying database tables from another instance of log.

What am I doing there?
These are my steps:
1. Install Cloudlog
2. Create new Admin
3. Create location
4. Create log
5. Link log with location

6. No statistics because of missing DXCC info, and then when I try to download it from ClubLog, it returns error after some time (5 minutes or so): "This request takes too long to process, it is timed out by the server. If it should not be timed out, please contact administrator of this web site to increase 'Connection Timeout'."

What am I doing wrong?

Regards,
Dragan, 4O4A

**Comment by 4O4A** [comment]
This is what I see last from this action. After that - nothing happens.

![image](https://user-images.githubusercontent.com/101504692/215254824-ffd1424f-b6d5-433e-84ec-4b7035b4a68c.png)

**Comment by 4O4A** [comment]
Some more info:

This is the result of the same steps when I do it locally, on my computer, executed in 5 seconds:

![image](https://user-images.githubusercontent.com/101504692/215280131-99173e4d-2f9d-450e-9945-6c70ee38a6cc.png)

I see locally that file cty.xml is downloaded from ClubLog.
On my web server, I don't see that file.
How I can check is there something on my server which is preventing access from ClubLog?
I have a cron job which sends data regularly to ClubLog, that works fine.

Is there anything else I can check on my server regarding this?

**Reply by phl0** [reply]
Probably an issue of permissions of the directory the cty.xml is to be stored to. Will check and let you know which dir it is.

**Reply by 4O4A** [reply]
I set 777 to "updates", still nothing -- showing "Downloading file...."  (from status.html)

**Reply by magicbug** [reply]
Likely Clublog blocking it or the servers not able to remote call it for some unknown reason.

I'll setup a fallback copy on my server to cover this I think, to help reduce these problems

**Reply by phl0** [reply]
Ok it should download to updates folder. If you set the permissions to 0777 the web server is able to write that file. So only reason left would be that the server cannot access clublog for whatever reason.

**Reply by phl0** [reply]
I have made a local copy to my web server for the cty.xml.gz. It seems that clublog is a little picky about access to theirs if you don't provide user agent etc. Simple wget/curl from command line for example failed.
@4O4A If you are up for testing with a different URL drop me a message.

**Comment by 4O4A** [comment]
This works, phl0. 
It seems that my server is not in good relationship with ClubLog when it comes to DXCC.
Oddly enough, upload of QSOs to ClubLog works OK.

**Reply by phl0** [reply]
I thought it might be an issue of missing user-agent or the like and re-wrote the code like this:
https://gist.github.com/phl0/a08e3056c7742d3adb061d42d2301694
But testing on @4O4A's instance still gives negative result. However pulling cty.xml.gz from my server works.

**Reply by magicbug** [reply]
Guessing the server IP is blocked? Is this a dedicated machine? maybe @g7vjr can shed some light.

**Comment by VolkerMue** [comment]
I had a different problem, or rather could not create a Station Location.

The manual dropping of the cty.xml did not help here.
Only with the Clublog patch and the automatic updates of the DXCC data I could successfully create the station location.

THX

**Comment by AlanFarley** [comment]
I had the same problem here, but I solved it by increasing the time limit in the file \application\controllers, line 169: set_time_limit(600);"

**Reply by AlanFarley** [reply]
file update.php
